frb_codegen/src/source_graph.rs

```diff
@@ -158,10 +166,41 @@ impl Debug for Enum {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct ImplTrait {
+    pub ident: Ident,
+    pub src: TypeImplTrait,
+    pub bound: TraitBound,
+}
+#[derive(Debug, Clone, Eq)]
+pub struct Impl {
+    pub self_ty: Ident,
+    pub trait_: Ident,
+}
```

Comment:
nit: would be great to add a few lines of comments discussing what this is. e.g.

```
/// This represents `impl {trait} for {self_ty}`
```

(not sure whether that is the case though, just an example)